lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1496,7 +1482,7 @@ static void checkTypeAccessibilityImpl(
   if (!typeAccessScope.hasValue())
     return;
 
-  if (!*typeAccessScope || *typeAccessScope == contextAccessScope)
+  if (typeAccessScope->isPublic() || *typeAccessScope == contextAccessScope)
```

Comment:
I need to go back over this section anyway: the specific thing it's supposed to check for isn't working. ([SR-2579](https://bugs.swift.org/browse/SR-2579)) I guess you're right: if there's no test case, we don't think it's doing anything important. (I think it may even just be an optimization.)